### PR TITLE
Make client deletion asynchronous

### PR DESCRIPTION
**Pull request by stefanwire**
## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://github.com/wireapp/wire-server/blob/develop/docs/developer/pr-guidelines.md)

### CODE REVIEW

**Review by smatting — APPROVED** [review]

File: services/brig/test/integration/API/User/Client.hs [hunk 1]
```diff
@@ -892,15 +892,15 @@ testMissingClient brig = do
...
+testAddMultipleTemporary :: Brig -> Galley -> Cannon -> Http ()
+testAddMultipleTemporary brig galley cannon = do
   uid <- userId <$> randomUser brig
   let clt1 =
         (defNewClient TemporaryClientType [somePrekeys !! 0] (someLastPrekeys !! 0))
           { newClientClass = Just PhoneClient,
             newClientModel = Just "featurephone1"
           }
-  _ <- addClient brig uid clt1
+  c <- responseJsonError =<< addClient brig uid clt1
```
Comment (elland) [inline services/brig/test/integration/API/User/Client.hs:903]: Can we rename this for clarity?

File: services/galley/test/integration/API/Teams/LegalHold.hs [hunk 2]
```diff
@@ -150,9 +151,16 @@ testsPublic s =
             [ test s "happy flow" testInWhitelist,
               test s "handshake between LH device and user with old clients is blocked" testOldClientsBlockDeviceHandshake,
               testGroup "no-consent" $
-                flip fmap [(a, b, c, d) | a <- [minBound ..], b <- [minBound ..], c <- [minBound ..], d <- [minBound ..]] $
-                  \args@(a, b, c, d) ->
-                    test s (show args) $ testNoConsentBlockOne2OneConv a b c d,
+                ( ( (,,,)
```
Comment (elland) [inline services/galley/test/integration/API/Teams/LegalHold.hs:154]: Nitpick: can we simplify this kind of thing to be easier to read / understand at a glance?

**Comment by elland** [comment]
Don't forget to hlint it? :D

File: services/galley/test/integration/API/Teams/LegalHold.hs [hunk 3]
```diff
@@ -150,9 +151,16 @@ testsPublic s =
...
+                      <$> (("connectFirst" :: String,) <$> [False, True])
+                      <*> (("teamPeer" :: String,) <$> [False, True])
+                      <*> (("approveLH" :: String,) <$> [False, True])
+                      <*> (("testPendingConnection" :: String,) <$> [False, True])
+                  )
```
Comment (pcapriotti) [inline services/galley/test/integration/API/Teams/LegalHold.hs:159]: Nice! You can also use `do` notation in the list monad to improve readability here:
```haskell
do connectFirst <- ("connectFirst",) <$> [False, True]
   ...
   let name = ...
   pure $ test s name $ ...
```

File: services/galley/test/integration/API/Teams/LegalHold.hs (hunk 2 above)
Comment (smatting) [reply services/galley/test/integration/API/Teams/LegalHold.hs]: Done. We used a list monad

File: services/galley/test/integration/API/Teams/LegalHold.hs (hunk 3 above)
Comment (smatting) [reply services/galley/test/integration/API/Teams/LegalHold.hs]: Ah, right, done

File: services/brig/src/Brig/IO/Intra.hs [hunk 4]
```diff
@@ -195,20 +195,27 @@ onPropertyEvent orig conn e =
...
+    HasRequestId m
+  ) =>
   -- | Originator of the event.
   UserId ->
   -- | Client connection ID.
   Maybe ConnId ->
   -- | The event.
   ClientEvent ->
-  (AppT r) ()
+  m ()
```
Comment (isovector) [inline services/brig/src/Brig/IO/Intra.hs:211]: Just as a heads up, changes like these make negative progress towards the polysemization effort.

Comment (smatting) [reply services/brig/src/Brig/IO/Intra.hs:211]: Ah, sorry, it was an oversight. I'll try not to introduce any mtl style code anymore

Comment (isovector) [reply services/brig/src/Brig/IO/Intra.hs:211]: Thanks! Greatly appreciated :)

**Pull request merged**